File: src/main/java/cz/forgottenempire/arma3servergui/system/ServerDetailsDto.java
package cz.forgottenempire.arma3servergui.system;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ServerDetailsDto {

    private long spaceLeft;
    private long spaceTotal;

    private long memoryLeft;
    private long memoryTotal;

    private double cpuUsage;
    private int cpuCount;

    private String osName;
    private String osVersion;
    private String osArchitecture;
}

File: frontend/src/dtos/ServerDetailsDto.ts
export interface ServerDetailsDto {
    spaceLeft: number,
    spaceTotal: number,
    memoryLeft: number,
    memoryTotal: number,
    cpuUsage: number,
    cpuCount: number,
    osName: string,
    osVersion: string,
    osArchitecture: string
}